processor/processor.go

```diff
@@ -1624,6 +1630,20 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 	reportMetrics = append(reportMetrics, validatedReportMetrics...)
 	// TRACKING PLAN - END
 
+	for sourceID, events := range groupedEventsBySourceId {
```

Comment:
`groupedEventsBySourceId` is used as an input for the tracking plan stage. After this stage, `validatedEventsBySourceId` is being used. Does it make sense to call the observers before the tracking plan stage?

Reply:
Makes sense, I changed the order to take place before tracking plan validation using `groupedEventsBySourceId`